hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/WriteBufferedP.java

```diff
@@ -56,10 +60,11 @@
 
     @Override
     public void init(@Nonnull Outbox outbox, @Nonnull Context context) {
-        buffer = createFn.apply(context);
-        if (buffer == null) {
+        B localBuff = createFn.apply(context);
+        if (localBuff == null) {
             throw new JetException("Null buffer created");
         }
+        buffer = (B) managedContext.initialize(localBuff);
```

Comment:
in `ProcessorSupplierWithService` there's a more detailed exception if the instance types don't match, maybe we could follow the same approach here as a standard?

Reply:
@gurbuzali are you able to address this in another PR? (also for other cases where use managed context), we can merge this one as is.

Reply:
sure, let's merge this one